client/landing/stepper/declarative-flow/internals/hooks/use-flow-navigation/index.tsx

```diff
@@ -62,10 +55,14 @@ export const useFlowNavigation = ( flow: Flow ): FlowNavigation => {
 			) {
 				// In-stepper auth.
 				if ( flow.__experimentalUseBuiltinAuth ) {
-					const signInPath = generatePath( `/:flow/:step/:lang?`, {
-						flow: flowName,
-						lang,
-						step: PRIVATE_STEPS.USER.slug,
+					const signInPath = createPath( {
+						pathname: generatePath( `/:flow/:step/:lang?`, {
+							flow: flowName,
+							lang,
+							step: PRIVATE_STEPS.USER.slug,
+						} ),
+						search: currentSearchParams.toString(),
+						hash: window.location.hash,
```

Comment:
Do we have cases we are using the hash?

Reply:
Not to my knowledge, but AI suggested it and it made sense to me.